Mathlib/Data/List/Sort.lean

```diff
@@ -192,6 +191,118 @@ alias ⟨_, _root_.Monotone.ofFn_sorted⟩ := sorted_le_ofFn_iff
 
 end Monotone
 
+end List
+
+open List
+
+namespace RelEmbedding
+
+variable {α β : Type*} {ra : α → α → Prop} {rb : β → β → Prop}
+
+@[simp]
+theorem sorted_listMap (e : ra ↪r rb) {l : List α} : (l.map e).Sorted rb ↔ l.Sorted ra := by
+  simp [Sorted, pairwise_map, e.map_rel_iff]
+
+@[simp]
+theorem sorted_swap_listMap (e : ra ↪r rb) {l : List α} :
+    (l.map e).Sorted (Function.swap rb) ↔ l.Sorted (Function.swap ra) := by
+  simp [Sorted, pairwise_map, e.map_rel_iff]
+
+end RelEmbedding
+
+namespace OrderEmbedding
+
+variable {α β : Type*} [Preorder α] [Preorder β]
+
+@[simp]
+theorem sorted_lt_listMap (e : α ↪o β) {l : List α} :
+    (l.map e).Sorted (· < ·) ↔ l.Sorted (· < ·) :=
```

Comment:
I'm not super happy with the namespacing here in the sense that `List.Sorted` becomes `sorted` and `List.map` becomes `listMap`, but I also believe this is the best option